torch/_functorch/aot_autograd.py

```diff
@@ -2611,6 +2641,15 @@ def create_synthetic_base_metadata(
         mutates_data = True if len(outer_indices) > 1 else m.input_info[outer_indices[0]].mutates_data
         mutates_metadata = False if len(outer_indices) > 1 else m.input_info[outer_indices[0]].mutates_metadata
         requires_grad = any(m.input_info[x].requires_grad for x in outer_indices)
+        mutations_hidden_from_autograd = all(m.input_info[x].mutations_hidden_from_autograd for x in outer_indices),
```

Comment:
Why are we doing this for all tensors to get the boolean? Shouldn't we need to make decision based on per input instead of querying all inputs?